src/internal/builder.ts

```diff
@@ -50,6 +51,20 @@ export class CFToolsClientBuilder {
         return this;
     }
 
+    /**
+     * The Enterprise API comes with special endpoint parameter for `/v1/users/resolve` (`client#resolve`), which allows to
+     * create a new CFTools account for an identity token that has never played on a CFTools Cloud-enabled server before.
+     * This functionality is not available to everyone with an Enterprise API token, but requires explicit access by CFTools.
+     * @param status Whether to enable the account creation API
+     */
+    public withAccountCreationAPI(status: boolean): CFToolsClientBuilder {
```

Comment:
A builder method on the client should control the basic behaviour of the client itself, but not how a specific API is used. While we could add a way in the client to enable/disable speciifc features, I don't like this approach for two reasons:
- it is not flexible: A user would need to build a new client when they want to resolve a specific SteamID, while automatically creating an account, and looking up another ID without automatically creating an account.
- it builds another abstraction and configuration into the client, which currently is only needed for a very specific use case.

I would therefore suggest an approach, which is more specific to the very specific use-case at hand:
- Add another type to the client#resolve method (e.g. `ResolveRequest`), which takes the player ID (either a `GenericId` or `{playerId: GenericId}` type) and the optional `autoCreateAccount` option, which is a boolean. This can either be done in a breaking or non-breaking change, both is possible. I would prefer a non-breaking change ofc :)
- Alternatively: Add a second optional parameter to client#resolve (e.g. `ResolveOptions`) which is an interface having one single, optional property `autoCreateAccount`
- based on the `autoCreateAccount` property, add the `create` URL parameter to the resolve request or not.

I would also be nice to have some additional error handling (like a distinct Error type when the provided Enterprise API Token does not have the Account Creation API enabled).

The only benefit of having this in the builder is the early error handling, that the account creation API is not available when a general-purpose API token is used, instead of an enterprise API token.
However, as the SDK is not an end-user visible thing, I think the caller of the SDK should ensure that the resolve method is only ever caller with the autoCreateAccount option, when an enterprise token is used. The caller knows both information and can handle this case on their end. If they don't they will get an error. Both it same, as it doesn't break the state of the system.

What do you think about my suggestion? :)

Reply:
> based on the autoCreateAccount property, add the create URL parameter to the resolve request or not.

This is only allowed **after** the initial request fails, aka. after it's been confirmed a CFTools account for the user doesn't exist - as included in the spec sent to your DMs on Discord :)

Everything is understandable, and I'll get it done!

Reply:
Addressed in https://github.com/FlorianSW/cftools-sdk/pull/15/commits/a195fae34b6438865dd0c4e8178221a8b010d002

Reply:
> This is only allowed after the initial request fails, aka. after it's been confirmed a CFTools account for the user doesn't exist

However, this is not enforced in the API, isn't it? 🤔 I would still make this a decision for the caller. We still might do two requests internally, but I don't see a big benefit in it :(